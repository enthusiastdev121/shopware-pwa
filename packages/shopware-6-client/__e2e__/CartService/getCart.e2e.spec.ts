import {
  getCart,
  addProductToCart,
  update,
} from "@shopware-pwa/shopware-6-client";

describe("shopware-6-client - E2E - CartService - getCart", () => {
  beforeEach(() => {
    update({ contextToken: "" });
  });

  it("should fetch cart for empty one", async () => {
    const result = await getCart();
    result.token = "mockedToken";
    expect(result).toMatchSnapshot();
  });

  it("should fetch cart with added product", async () => {
    await addProductToCart("044a190a54ab4f06803909c3ee8063ef", 2);
    const result = await getCart();
    result.token = "mockedToken";
    expect(result).toMatchSnapshot();
  });
});
