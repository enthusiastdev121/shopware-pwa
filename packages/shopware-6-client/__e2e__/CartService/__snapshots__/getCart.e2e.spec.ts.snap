// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`shopware-6-client - E2E - CartService - getCart should fetch cart for empty one 1`] = `
Object {
  "deliveries": Array [],
  "errors": Array [],
  "extensions": Array [],
  "lineItems": Array [],
  "modified": false,
  "name": "8a243080f92e4c719546314b577cf82b",
  "price": Object {
    "calculatedTaxes": Array [],
    "extensions": Array [],
    "netPrice": 0,
    "positionPrice": 0,
    "taxRules": Array [],
    "taxStatus": "gross",
    "totalPrice": 0,
  },
  "token": "mockedToken",
  "transactions": Array [
    Object {
      "amount": Object {
        "calculatedTaxes": Array [],
        "extensions": Array [],
        "quantity": 1,
        "referencePrice": null,
        "taxRules": Array [],
        "totalPrice": 0,
        "unitPrice": 0,
      },
      "extensions": Array [],
      "paymentMethodId": "cad2356ee68c473ca7da517d0ad32de1",
    },
  ],
}
`;

exports[`shopware-6-client - E2E - CartService - getCart should fetch cart with added product 1`] = `
Object {
  "deliveries": Array [],
  "errors": Array [],
  "extensions": Array [],
  "lineItems": Array [],
  "modified": false,
  "name": "8a243080f92e4c719546314b577cf82b",
  "price": Object {
    "calculatedTaxes": Array [],
    "extensions": Array [],
    "netPrice": 0,
    "positionPrice": 0,
    "taxRules": Array [],
    "taxStatus": "gross",
    "totalPrice": 0,
  },
  "token": "mockedToken",
  "transactions": Array [
    Object {
      "amount": Object {
        "calculatedTaxes": Array [],
        "extensions": Array [],
        "quantity": 1,
        "referencePrice": null,
        "taxRules": Array [],
        "totalPrice": 0,
        "unitPrice": 0,
      },
      "extensions": Array [],
      "paymentMethodId": "cad2356ee68c473ca7da517d0ad32de1",
    },
  ],
}
`;
